### PR TITLE
Extended example of parameters.

### LINKED ISSUE #1203: How to support hyphen in urls?
(I'd ask this on riot/matrix but couldn't find the room for some reason)

If we already have a query parameter unkind-rust-name=foo, is there a way to encode this as a route? 

```rust
#[get(/bar?unkind-rust-name=<kind_rust_name>)]
fn bar(kind_rust_name: String) {}
```

Many thanks in advance!

### PR DESCRIPTION
**Pull request by gilescope**
Shows how to handle parameter renaming for the next person that is looking for it (Closes #1203 )

### CODE REVIEW

**Comment by jebrosen** [comment]
Tests are failing because the expected name was renamed but the tests weren't changed to match.

**Comment by gilescope** [comment]
Ah sorry didn't spot the tests. Will fixup.

**Comment by gilescope** [comment]
Great backtrace doesn't compile on an azure windows box. Not sure what I can do about that?

**Comment by jebrosen** [comment]
The Windows CI failure is caused by `mysql 17` -> `mysql_common 0.19.2` -> `lexical 4` -> `cfg-if = 0.1.9`. This breaks `backtrace 0.3.42`, which should say it depends on `cfg-if ^0.1.10` but did not. Since the version correction appears to be the only thing changed in `backtrace 0.3.43`, I will ask if yanking `0.3.42` would be possible.

File: README.md [hunk 1]
```diff
@@ -158,3 +158,4 @@ Rocket is licensed under either of the following, at your option:
  * MIT License ([LICENSE-MIT](LICENSE-MIT) or http://opensource.org/licenses/MIT)
 
 The Rocket website source is licensed under [separate terms](site#license).
+
```
Comment (SergioBenitez) [inline README.md]: This shouldn't be here.

File: examples/query_params/src/main.rs [hunk 2]
```diff
@@ -8,6 +8,7 @@ use rocket::request::{Form, LenientForm};
 
 #[derive(FromForm)]
 struct Person {
+    #[form( field = "first-name" )]  // renaming parameter to valid rust identifier
```
Comment (SergioBenitez) [inline examples/query_params/src/main.rs]:
```suggestion
    /// Use the `form` attribute to expect an invalid Rust identifier in the HTTP form.
    #[form(field = "first-name")]
```

**Comment by gilescope** [comment]
@SergioBenitez all good?

**Comment by gilescope** [comment]
@SergioBenitez I can't see what changes are outstanding? What's left to merge this?

**Comment by gilescope** [comment]
Great, @SergioBenitez can you merge in please? I don't have permissions to do that bit?

**Comment by SergioBenitez** [comment]
Merged in ff4b63c0be85c602feb4d8426a5c777db7ad6d87. Thank you!

**Pull request closed without merging**